app/services/additional_duty_applicable_measures_merger.rb

```diff
@@ -0,0 +1,21 @@
+class AdditionalDutyApplicableMeasuresMerger
+  include CommodityHelper
+
+  def call
+    send("#{user_session.commodity_source}_applicable_measures")
+  end
+
+  private
+
+  def xi_applicable_measures
+    filtered_commodity.applicable_measures.select(&:applicable?).concat(uk_filtered_commodity.excise_measures)
+  end
+
+  def uk_applicable_measures
+    filtered_commodity.applicable_measures.select(&:applicable?)
+  end
+
+  def user_session
```

Comment:
Planning on rejigging this so it isn't needed as much - there seems to be a significant crossover with the commodity helper so I can move this method there and kill it in like 5 other places